homebridge/thermostat.ts

```diff
@@ -0,0 +1,189 @@
+          )
+          return
+        }
+        this.logger.info(`Setting ${this.device.name} mode to ${mode}`)
+
+        return this.device.setInfo({ device: { v1: { mode } } })
+      },
+    })
+    // Only allow 'TargetHeatingCoolingState's which can be mapped to Ring modes
+    // Specifically, this omits .AUTO
+    this.getService(Service.Thermostat)
+      .getCharacteristic(Characteristic.TargetHeatingCoolingState)
+      .setProps({
+        validValues: [
+          Characteristic.TargetHeatingCoolingState.OFF,
+          Characteristic.TargetHeatingCoolingState.HEAT,
+          Characteristic.TargetHeatingCoolingState.COOL,
+        ],
+      })
+
+    this.registerCharacteristic({
+      characteristicType: Characteristic.CurrentTemperature,
+      serviceType: Service.Thermostat,
+      getValue: async () => {
+        const temperature = await this.getTemperatureFromComponentDevices()
+        if (!temperature) {
+          return
+        }
+        // Documentation: https://developers.homebridge.io/#/characteristic/CurrentTemperature
+        // 'Characteristic.CurrentTemperature' supports 0.1 increments
+        return Math.round(temperature * 10) / 10
+      },
+    })
+
+    this.registerCharacteristic({
+      characteristicType: Characteristic.TargetTemperature,
+      serviceType: Service.Thermostat,
+      getValue: ({ setPoint }) => {
+        return setPoint
+      },
+      setValue: (setPoint: number) => {
+        this.logger.info(
+          `Setting ${this.device.name} target temperature to ${setPoint}`
+        )
+
+        // Documentation: https://developers.homebridge.io/#/characteristic/TargetTemperature
+        // 'Characteristic.TargetTemperature' has a valid range from 10 to 38 degrees celsius,
+        // but devices may support a different range. When limits differ, accept the more strict.
+        const setPointMin = Math.max(this.device.data.setPointMin || 10, 10),
+          setPointMax = Math.min(this.device.data.setPointMax || 38, 38)
+
+        if (setPoint < setPointMin || setPoint > setPointMax) {
+          this.logger.error(
+            `Ignoring request to set ${this.device.name} target temperature to ${setPoint}. Target temperature must be between ${setPointMin} and ${setPointMax}.`
+          )
+          return
+        }
+
+        return this.device.setInfo({ device: { v1: { setPoint } } })
+      },
+    })
+
+    this.registerCharacteristic({
+      characteristicType: Characteristic.TemperatureDisplayUnits,
+      serviceType: Service.Thermostat,
+      getValue: () => {
+        // Neither thermostats nor their component devices (e.g. temperature sensors)
+        // appear to include the unit preference. Hardcoding Fahrenheit as the default.
+        return Characteristic.TemperatureDisplayUnits.FAHRENHEIT
+      },
+      setValue: () => {
+        // noop
+        // Setting display unit is unsupported
```

Comment:
Might be better to limit the `validValues` here since we aren't supporting Celcius

Reply:
I attempted to limit the values in https://github.com/dgreif/ring/pull/331/commits/7d8eca5362c2c69e04ed24a667513931e2d4aff0, but both display unit values are still selectable in the Home app:

![IMG_3114](https://user-images.githubusercontent.com/3104489/82754585-bf1c0e00-9d9b-11ea-8ac0-b87e692cfb48.jpeg)